lib/ecto/repo.ex

```diff
@@ -74,8 +77,9 @@ defmodule Ecto.Repo do
         Ecto.Repo.Config.config(@otp_app, __MODULE__)
       end
 
-      def start_link do
-        @adapter.start_link(__MODULE__, config())
+      def start_link(custom_config \\ []) do
```

Comment:
I have pushed a different implementation to master to allow pool customization. We now just need the `start_link/1` changes plus guarantee the name and pool given to `start_link/1` have higher priority than the one in `__pool__`. For that, we should have tests.